splice_machine/src/test/java/com/splicemachine/pl/PL_SaveSourceCode_IT.java

```diff
@@ -0,0 +1,125 @@
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.splicemachine.pl;
+
+import com.splicemachine.derby.test.framework.SpliceSchemaWatcher;
+import com.splicemachine.derby.test.framework.SpliceWatcher;
+import com.splicemachine.derby.test.framework.TestConnection;
+import com.splicemachine.test.SerialTest;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.spark_project.guava.collect.Lists;
+
+import javax.sql.rowset.serial.SerialBlob;
+import java.sql.Blob;
+import java.sql.CallableStatement;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.util.Collection;
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * Tests saving/deleting code objects from SYSSOURCECODE table.
+ */
+@Category(value = {SerialTest.class})
+@RunWith(Parameterized.class)
+public class PL_SaveSourceCode_IT {
+
+    private static final String SCHEMA = PL_SaveSourceCode_IT.class.getSimpleName();
+
+    @ClassRule
+    public static SpliceSchemaWatcher spliceSchemaWatcher = new SpliceSchemaWatcher(SCHEMA);
+
+    @Rule
+    public SpliceWatcher methodWatcher = new SpliceWatcher(SCHEMA);
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        Collection<Object[]> params = Lists.newArrayListWithCapacity(2);
+        params.add(new Object[]{"jdbc:splice://localhost:1527/splicedb;create=true;user=splice;password=admin"});
+        params.add(new Object[]{"jdbc:splice://localhost:1527/splicedb;create=true;user=splice;password=admin;useSpark=true"});
+        return params;
+    }
+
+    private String connectionString;
+
+    public PL_SaveSourceCode_IT(String connecitonString) {
+        this.connectionString = connecitonString;
+    }
+
+    @Before
+    public void createTables() throws Exception {
+        Connection conn = new TestConnection(DriverManager.getConnection(connectionString, new Properties()));
+        conn.setSchema(SCHEMA.toUpperCase());
+        methodWatcher.setConnection(conn);
+    }
+
+    @Test
+    public void saveSourceCode() throws Exception {
+        // no source code has been saved yet, so following query should return 0 rows
+        assertEquals(0, methodWatcher.queryList("SELECT SOURCE_CODE FROM SYS.SYSSOURCECODE").size());
+
+        // save some source code.
+        String SCHEMA_NAME = "not_necessarily_a_valid_db_schema";
+        String OBJECT_NAME = "identity";
+        String OBJECT_TYPE = "SpliceSQLFunction";
+        String OBJECT_FORM = "SpliceSQL:1.0@UTF8";
+        String SOURCE_CODE = "select ? from sysibm.sysdummy1";
```

Comment:
They are user-defined, meaningful only to the consumer of the source code data.  This is only an  example of what a user might find useful to define.